README.md

```diff
@@ -137,26 +166,41 @@ The Azure Key Vault Provider offers four modes for accessing a Key Vault instanc
 
 > Supported with linux and windows
 
-1. Add your service principal credentials as a Kubernetes secrets accessible by the Secrets Store CSI driver.
+1. Add your service principal credentials as a Kubernetes secrets accessible by the Secrets Store CSI driver. If using AKS you can learn about [service principals in AKS here.](https://docs.microsoft.com/azure/aks/kubernetes-service-principal)
 
     ```bash
+    # Client ID will be the App ID of your service principal
+    # Client Secret will be the Password of your service principal
+
     kubectl create secret generic secrets-store-creds --from-literal clientid=<CLIENTID> --from-literal clientsecret=<CLIENTSECRET>
     ```
 
     Ensure this service principal has all the required permissions to access content in your Azure key vault instance.
-    If not, you can run the following using the Azure cli:
+    If you do not have a service principal, you can run the following Azure CLI commands to create a new service principal and assign permissions:
 
     ```bash
-    # Assign Reader Role to the service principal for your keyvault
-    az role assignment create --role Reader --assignee <principalid> --scope /subscriptions/<subscriptionid>/resourcegroups/<resourcegroup>/providers/Microsoft.KeyVault/vaults/<keyvaultname>
+    # Set environment variables
+    SPNAME=<servicePrincipalName>
+    APPID=<service-principal-appId>
```

Comment:
can we replace this with 
```
APPID=$(az ad sp show --id http://${SPNAME} --query appId -otsv)
```

Reply:
good idea, added and moved the sp creation step to first step